vault/logical_system.go

```diff
@@ -3386,25 +3398,34 @@ func (b *SystemBackend) pathInternalUIMountRead(ctx context.Context, req *logica
 		fullMountPath = ns.Path + me.Namespace().Path + me.Path
 	}
 
+	acl, err := b.verifyAuthorizedMountAccess(ctx, req)
+	if err != nil {
+		return errResp, err
+	}
+
+	if !hasMountAccess(ctx, acl, fullMountPath) {
+		return errResp, logical.ErrPermissionDenied
+	}
+
+	return resp, nil
+}
+
+func (b *SystemBackend) verifyAuthorizedMountAccess(ctx context.Context, req *logical.Request) (*ACL, error) {
```

Comment:
The name of this method is a bit misleading since I don't think it has anything to do with mounts.